subsys/net/lib/softap_wifi_provision/softap_wifi_provision.c

```diff
@@ -0,0 +1,1187 @@
+		break;
+	case NET_EVENT_WIFI_AP_STA_DISCONNECTED:
+		LOG_DBG("NET_EVENT_WIFI_AP_STA_DISCONNECTED");
+
+		print_mac("Client STA disconnected, MAC: ", cb->info);
+
+		notify_app(SOFTAP_WIFI_PROVISION_EVT_CLIENT_DISCONNECTED);
+		break;
+	case NET_EVENT_WIFI_SCAN_RESULT:
+		wifi_scan_result_handle(cb);
+		break;
+	case NET_EVENT_WIFI_SCAN_DONE:
+		LOG_DBG("NET_EVENT_WIFI_SCAN_DONE");
+
+		/* Scan results cached, waiting for client to request available networks and
+		 * provide Wi-Fi credentials.
+		 */
+		new_event(EVENT_SCAN_DONE);
+		break;
+	default:
+		break;
+	}
+}
+
+static int wifi_scan(void)
+{
+	int ret;
+	struct net_if *iface = net_if_get_first_wifi();
+	struct wifi_scan_params params = { 0 };
+
+	LOG_DBG("Scanning for Wi-Fi networks...");
+
+	/* Clear scan result buffer */
+	memset(scan_result_buffer, 0, sizeof(scan_result_buffer));
+
+	ret = net_mgmt(NET_REQUEST_WIFI_SCAN, iface, &params, sizeof(params));
+	if (ret) {
+		LOG_ERR("Failed to start Wi-Fi scan, error: %d", ret);
+		return -ENOEXEC;
+	}
+
+	return 0;
+}
+
+static int ap_enable(void)
+{
+	int ret;
+	struct net_if *iface = net_if_get_first_wifi();
+	static struct wifi_connect_req_params params = {
+		.timeout = SYS_FOREVER_MS,
+		.band = WIFI_FREQ_BAND_UNKNOWN,
+		.channel = WIFI_CHANNEL_ANY,
+		.security = WIFI_SECURITY_TYPE_NONE,
+		.mfp = WIFI_MFP_OPTIONAL,
+		.ssid = CONFIG_SOFTAP_WIFI_PROVISION_SSID,
+		.ssid_length = strlen(CONFIG_SOFTAP_WIFI_PROVISION_SSID)
+	};
+
+	ret = net_mgmt(NET_REQUEST_WIFI_AP_ENABLE, iface, &params,
+		       sizeof(struct wifi_connect_req_params));
+	if (ret) {
+		LOG_ERR("Failed to enable AP, error: %d", ret);
+		return -ENOEXEC;
+	}
+
+	return 0;
+}
+
+static int ap_disable(void)
+{
+	int ret;
+	struct net_if *iface = net_if_get_first_wifi();
+
+	ret = net_mgmt(NET_REQUEST_WIFI_AP_DISABLE, iface, NULL, 0);
+	if (ret) {
+		LOG_ERR("Failed to disable AP, error: %d", ret);
+		return -ENOEXEC;
+	}
+
+	return 0;
+}
+
+static int parse_and_store_credentials(const char *body, size_t body_len)
+{
+	int ret;
+	uint32_t flag;
+	WifiConfig creds = WifiConfig_init_zero;
+
+	pb_istream_t stream = pb_istream_from_buffer((const pb_byte_t *)body, body_len);
+
+	if (!pb_decode(&stream, WifiConfig_fields, &creds)) {
+		LOG_ERR("Decoding credentials failed");
+		return -1;
+	}
+
+	/* Convert BSSID to readable format */
+	char bssid_str[sizeof("xx:xx:xx:xx:xx:xx")];
+
+	net_sprint_ll_addr_buf(creds.wifi.bssid.bytes, creds.wifi.bssid.size,
+			       bssid_str, sizeof(bssid_str));
+
+	LOG_DBG("Received Wi-Fi credentials: "
+		"ssid: %.*s, bssid: %s, passphrase: xxxxxx, sectype: %d, channel: %d, band: %d",
+		creds.wifi.ssid.size, creds.wifi.ssid.bytes,
+		bssid_str,
+		creds.wifi.auth,
+		creds.wifi.channel,
+		creds.wifi.band);
+
+	enum wifi_security_type sec_type =
+		creds.wifi.auth == AuthMode_WPA_WPA2_PSK ? WIFI_SECURITY_TYPE_PSK :
+		creds.wifi.auth == AuthMode_WPA2_PSK ? WIFI_SECURITY_TYPE_PSK_SHA256 :
```

Comment:
WPA2-PSK and WPA2-PSK-SHA256 are two different options, so, might need to rename it accordingly

WPA_WPA2_PSK->WPA2_PSK
WPA2_PSK->WPA2_PSK_SHA256

Reply:
The naming schemas has been adopted from the BLE wifi prov lib/proto. We decided to keep them as similar as possible.
The naming is a bit bland yes.